src/cdk/table/table.ts

```diff
@@ -666,7 +673,20 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
     });
 
     this._updateNoDataRow();
-    this.updateStickyColumnStyles();
+
+    // Allow the new row data to render before measuring it.
+    // TODO: Remove indirection once _ngZone is required.
+    const deferredUpdateStickyColumnStyles = () => {
+      setTimeout(() => {
```

Comment:
Would `ngZone.onStable` work here?

Reply:
Also potentially `requestAnimationFrame` instead of `setTimeout` if `onStable` doesn't work.

Reply:
onStable works swimmingly. Thanks for the suggestion.